### DISCUSSION TITLE
How to work with direct json data using yq commands

### DISCUSSION THREAD

**Discussion by dnyaneshwarjadhav** [mikefarah/yq#1249] [Q&A]
I am having json file 
sample.yaml
{"cat": "meow"}

Following query works
yq '.a' sample.yml

But If I want to store json data into variable then how to perform query ?
$ yq '. | length' "a: cat"
Error: open a: cat: The system cannot find the path specified.

**Comment by mikefarah** [comment]
I'm not sure what you mean by 'store json data into variable' - what's the result you want?

**Comment by balki** [comment]
Do you mean the yaml string is in a variable instead of `sample.yaml`? If so, below works
```
❯ yq '. | length' <<<"a: cat"
1
❯ ystr="a: cat"              
❯ yq '. | length' <<<$ystr   
1
❯ yq '.a' <<<$ystr
cat
```
